colcon_core/task/python/build.py

```diff
@@ -133,30 +86,40 @@ def __init__(self):  # noqa: D107
 
             # invoke `setup.py develop` step in build space
             # to avoid placing any files in the source space
+
+            # --editable causes this to skip creating/editing the
+            # easy-install.pth file
             cmd = [
                 executable, 'setup.py',
                 'develop', '--prefix', args.install_base,
+                '--editable', '--build-directory', os.path.join(
+                    args.build_base, 'build'),
                 '--no-deps',
             ]
             if setup_py_data.get('data_files', []):
                 cmd += ['install_data', '--install-dir', args.install_base]
             self._append_install_layout(args, cmd)
-            async with _EasyInstallPthLockAsyncContext(pkg, args):
-                rc = await check_call(
-                    self.context, cmd, cwd=args.build_base, env=env)
+            rc = await check_call(
+                self.context, cmd, cwd=args.build_base, env=env)
             if rc and rc.returncode:
                 return rc.returncode
 
-            # explicitly add the build directory to the PYTHONPATH
-            # to maintain the desired order
-            # otherwise the path from the easy-install.pth (which is the build
-            # directory) will be added to the PYTHONPATH implicitly
-            # but behind potentially other directories from the PYTHONPATH
-            if additional_hooks is None:
-                additional_hooks = []
-            additional_hooks += create_environment_hook(
-                'pythonpath_develop', Path(args.build_base), pkg.name,
-                'PYTHONPATH', args.build_base, mode='prepend')
+            egg_link = os.path.join(python_lib,
+                pkg.name + '.egg-link')
+
+            with open(egg_link, 'r') as h:
+                lines = h.readlines()
+            link_target = Path(lines[0])
+
+            link_source = Path(python_lib, pkg.name)
```

Comment:
This assumes that a package contains a single directory named after the package. While that is commonly the case what if a package diverges from that - either having multiple or using a different name?

Reply:
Yeah, I haven't quiiiiite figured out the proper way to symlink this in. This PR is still very WIP.

The intent of that line (which doesn't work) is to link the *egg* not the modules themselves, but I haven't figured out the most elegant way yet to do this.